### PR TITLE
redirect: Add client support for browser-redirect interactor

### PR DESCRIPTION
**Pull request by mhilton**

### CODE REVIEW

**Comment by mhilton** [comment]
Note this code was copied, without modification, from https://github.com/CanonicalLtd/candid

File: redirect/redirect_test.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+// Package redirect implements redirection based login.
```
Comment (frankban) [inline redirect/redirect_test.go]: This seems to be copy/paste leftover.

File: redirect/redirect_test.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	// Fake an empty InteractionRequiredError
+	irerr.Info = &httpbakery.ErrorInfo{}
+
+	_, err = i.Interact(ctx, nil, "", irerr)
+	c.Assert(errgo.Cause(err), qt.Equals, httpbakery.ErrInteractionMethodNotFound)
+
+	redirect.SetInteraction(irerr, "https://www.example.com/login", "https://www.example.com/token")
+
+	_, err = i.Interact(ctx, nil, "", irerr)
+	c.Assert(httpbakery.IsInteractionError(err), qt.Equals, true, qt.Commentf("%s", err))
```
Comment (frankban) [inline redirect/redirect_test.go]: Replace this with `c.Assert(err, qt.Satisfies, httpbakery.IsInteractionError)` ?

File: redirect/redirect_test.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+	ierr := err.(*httpbakery.InteractionError)
+	c.Assert(redirect.IsRedirectRequiredError(ierr.Reason), qt.Equals, true)
```
Comment (frankban) [inline redirect/redirect_test.go]: Similar suggestion about using qt.Satisfies.

**Review by frankban — APPROVED** [review]
Looks good with a couple of minor suggestions. Thanks!

File: redirect/redirect_test.go (hunk 3 above)
Comment (mhilton) [reply redirect/redirect_test.go]: done

File: redirect/redirect_test.go (hunk 2 above)
Comment (mhilton) [reply redirect/redirect_test.go]: done

File: redirect/redirect_test.go (hunk 1 above)
Comment (mhilton) [reply redirect/redirect_test.go]: done

**Pull request merged**